WordPress/Classes/Utility/WebKitViewController.swift

```diff
@@ -187,7 +187,13 @@ class WebKitViewController: UIViewController, WebKitAuthenticatable {
 
         NSLayoutConstraint.activate(edgeConstraints)
 
-        view.pinSubviewAtCenter(stackView)
+        // we are pinning the top and bottom of the stack view to the safe area to prevent unintentionally hidden content/overlaps (ie cookie acceptance popup) then let the rest stay centered to allow the reader website background to occupy all available space, and to allow the site preview to show a user's own site without stretching it
+        let safeArea = self.view.safeAreaLayoutGuide
+
+        stackView.topAnchor.constraint(equalTo: safeArea.topAnchor).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: safeArea.bottomAnchor).isActive = true
+        stackView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        stackView.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
```

Comment:
👋 @kduncanwelke - this looks really good! Thanks for your work so far.

This screenshot got me thinking:

<img width="500px" src="https://user-images.githubusercontent.com/43656898/160196747-f79cabb6-9e6e-4ca2-814f-154dc570d6ef.png" />

That little gray bar at the bottom is from the UIStackView obeying the bottom safe area. In cases like these we may prefer to allow the web content to go behind the Home Indicator, so I wonder if we can:

1. Use the view's safe area bottom constraint when there's a toolbar
2. Use the view's bottom constraint when there isn't one

This will add a little bit of complexity, but may give us the best of both worlds:

1. Remove the `centerYAnchor` constraint, but keep the `centerXAnchor`. I can't think of a reason why it's needed, but I may be wrong here. It seems that if the top and bottom anchors are set at all times we don't need it. I do understand why we need the `centerXAnchor`, and that's for when the UIStackView's width gets resized.
3. Create a property that stores our safe area bottom constraint so that we can alter its priority depending on if there's a toolbar or not.
4. On `viewWillLayoutSubviews()` detect whether or not the toolbar is showing, and decided what priority to set the bottom constraint to.

[I've pushed up a branch](https://github.com/wordpress-mobile/WordPress-iOS/compare/temp/webkitvc-safe-areas) of what I'm thinking, but would love to hear your thoughts. Thanks!

Reply:
I'll try out these changes and see if I can get that bar to disappear - I'd like to see this through to completion for the experience and because of the time invested so far. Thanks for checking it over again!